src/ingest-pipeline/airflow/dags/standardize_extensions.py

```diff
@@ -0,0 +1,183 @@
+import os
+from pathlib import Path
+from pprint import pprint
+from datetime import datetime, timedelta
+import logging
+
+from airflow.configuration import conf as airflow_conf
+from airflow.operators.python import PythonOperator
+from airflow.exceptions import AirflowException
+
+import utils
+from utils import (
+    localized_assert_json_matches_schema as assert_json_matches_schema,
+    HMDAG,
+    get_queue_resource,
+    )
+
+# Following are defaults which can be overridden later on
+default_args = {
+    'owner': 'hubmap',
+    'depends_on_past': False,
+    'start_date': datetime(2019, 1, 1),
+    'email': ['joel.welling@gmail.com'],
+    'email_on_failure': False,
+    'email_on_retry': False,
+    'retries': 1,
+    'retry_delay': timedelta(minutes=1),
+    'xcom_push': True,
+    'queue': get_queue_resource('validation_test'),
+}
+
+"""
+Examples:
+    - Bare-bones config that will change ".tif" extensions
+      to ".tiff" for a single upload:
+        {"uuid_list": [<uuid>]}
+    - Config that will do the same for multiple uploads:
+        {"uuid_list": [<uuid1>, <uuid2>]}
+    - Config with custom extension_pair, find but not replace:
+        {
+            "uuid_list": [<uuid1>],
+            "extension_pair": {
+                "target": ".jpg",
+                "replacement": ".jpeg"
+            }
+            "find_only": true
+        }
+"""
+with HMDAG('standardize_extensions',
+           schedule_interval=None,
+           is_paused_upon_creation=False,
+           default_args=default_args,
+           ) as dag:
+
+    app_client_secret = airflow_conf.as_dict().get('connections', {}).get('APP_CLIENT_SECRET')
+    assert type(app_client_secret) is str
+    op_kwargs = {'crypt_auth_tok': utils.encrypt_tok(app_client_secret).decode()}
+
+    def get_single_dir_path(uuid: str, **kwargs):
+        ds_rslt = utils.pythonop_get_dataset_state(
+            dataset_uuid_callable=lambda **kwargs: uuid,
+            **kwargs
+        )
+        # This will fail the DAG; could be written to log and skip any UUIDs with issues
+        if not ds_rslt:
+            raise AirflowException(f"Invalid uuid/doi for group: {uuid}")
+
+        for key in ["entity_type", "status", "uuid", "local_directory_full_path"]:
+            assert key in ds_rslt, f"Dataset status for {uuid} has no {key}"
+
+        if not ds_rslt["entity_type"] in ["Upload"]:
+            raise AirflowException(f"Entity {uuid} is not an Upload")
+
+        if not ds_rslt["status"] in ["New", "Invalid", "Error"]:
```

Comment:
This is fine for now, but we may want to extend the status set based on feedback from Brendan.

Reply:
Oh right, yes, I just guessed here, I meant to check! Happy to update in future.